src/devices/bus/spectrum/wafa.cpp

```diff
@@ -0,0 +1,187 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+/**********************************************************************
+
+    Rotronics Wafadrive
+
+	Provides 2 built-in 'infinite loop' tape based drives that act like
+	discs (similar to Microdrive), a Centronics port, a RS232 port, and
+	the standard pass-thru connector.
+
+	Tapes could hold 128k, 64k or 16k (approximate sizes) with longer
+	seek times on the higher capacity media.  Loading was significantly
+	slower than the Microdrive (but faster than cassette)
+
+	Media was highly unreliable and prone to snapping.
+
+	use
+	NEW *
+	to initialize
+
+**********************************************************************/
+
+#include "emu.h"
+#include "wafa.h"
+
+//**************************************************************************
+//  DEVICE DEFINITIONS
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(SPECTRUM_WAFA, spectrum_wafa_device, "wafadrive", "Rotronics Wafadrive")
```

Comment:
Could you rename the romset to "spectrum_wafa" to be consistent with other Spectrum devices?

Reply:
done, btw is there unsubmitted code for
//device.option_add("disciple", SPECTRUM_DISCIPLE);
and
//device.option_add("plusd", SPECTRUM_PLUSD);

it's very confusing to see commented out entries where no code seems to exist for them

Reply:
Yes, but what I have for those devices is little more than a ROM placeholder. I don't mind if they're removed or you decide to implement them. Otherwise I'll get back to them eventually.

Reply:
Well a ROM placeholder is better than nothing, it means nobody has to start over if there is an interest.